npm/test/s3upload.test.ts

```diff
@@ -160,18 +164,62 @@ test("a single file upload returns the correct key and should call addFileStatus
     jest.fn(),
     ""
   )
-  expect(fetchMock).toHaveBeenCalledWith("/add-file-status", {"body": "{\"fileId\":\"1df92708-d66b-4b55-8c1e-bb945a5c4fb5\",\"statusType\":\"Upload\",\"statusValue\":\"Failed\"}", "credentials": "include", "headers": {"Content-Type": "application/json", "Csrf-Token": "abcde", "X-Requested-With": "XMLHttpRequest"}, "method": "POST"})
+  expect(fetchMock).toHaveBeenCalledWith("/add-file-status", {
+    body: '{"fileId":"1df92708-d66b-4b55-8c1e-bb945a5c4fb5","statusType":"Upload","statusValue":"Failed"}',
+    credentials: "include",
+    headers: {
+      "Content-Type": "application/json",
+      "Csrf-Token": "abcde",
+      "X-Requested-With": "XMLHttpRequest"
+    },
+    method: "POST"
+  })
 
   const input = s3Mock.calls().pop()!.args[0].input as { Key: string }
   expect(input.Key).toEqual(
     `${userId}/16b73cc7-a81e-4317-a7a4-9bbb5fa1cc4e/1df92708-d66b-4b55-8c1e-bb945a5c4fb5`
   )
 })
 
+test("a single file upload returns an error if it fails", async () => {
+  const tdrFileWithPath = createTdrFile({
+    fileId: "1df92708-d66b-4b55-8c1e-bb945a5c4fb5"
+  })
+  fetchMock.mockResponse(
+    JSON.stringify({
+      fileId: tdrFileWithPath.fileId,
+      statusType: "Upload",
+      statusValue: "Failed"
+    })
+  )
+  s3Mock.on(PutObjectCommand).resolves({ $metadata: { httpStatusCode: 500 } })
+  const s3Upload = new S3Upload(s3Mock as unknown as S3Client, "")
+
+  const result:
+    | {
```

Comment:
This isn't directly related to your changes but this is the third place now that we're writing the type out like this. Do you think it would be clearer to create an interface in `s3Upload/index.ts` like
```
interface IUploadResult {
  sendData: ServiceOutputTypes[]
  processedChunks: number
  totalChunks: number
}
```
Then replace this type and the one in `s3Upload/index.ts` and the one in `clientfileprocessing.test.ts` It doesn't really change anything but it might look a bit neater. What do you think?